pkg/platform/abstract/platform.go

```diff
@@ -433,6 +446,130 @@ func (ap *Platform) ResolveReservedResourceNames() []string {
 	}
 }
 
+// CleanseProjects will filter out some projects
+func (ap *Platform) CleanseProjects(cleanseOptions *platform.CleanseOptions,
+	projects []platform.Project) ([]platform.Project, error) {
+
+	// no cleansing is mandated
+	if len(cleanseOptions.MemberIds) == 0 {
+		return projects, nil
+	}
+
+	appendLock := sync.Mutex{}
+	errGroup, _ := errgroup.WithContext(context.TODO(), ap.Logger)
+	var permittedProjects []platform.Project
+	for _, projectInstance := range projects {
+		projectInstance := projectInstance
+		errGroup.Go("QueryOPAProjectPermissions", func() error {
+
+			// Check OPA permissions
+			if allowed, err := ap.QueryOPAProjectPermissions(projectInstance.GetConfig().Meta.Name,
+				opa.ActionRead,
+				cleanseOptions.MemberIds,
+				cleanseOptions.RaiseForbidden); err != nil {
+				return errors.Wrap(err, "Failed authorizing OPA permissions for resource")
+			} else if allowed {
+				appendLock.Lock()
+				permittedProjects = append(permittedProjects, projectInstance)
+				appendLock.Unlock()
```

Comment:
will be much nicer if we add a synced slice struct - struct that holds a slice and a lock, and exposes append (does the lock/unlock with defer, safely) and returning the underlining slice

Reply:
when []interface to []<concrete> would be possible, until then kiss it